File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int lastPos[128];
        for (int i = 0; i< 128; i++)
        {
            lastPos[i] = -1;
        }

        int left = 0, maxLen = 0;

        for(int right = 0; right < s.size(); right++){
            if(lastPos[s[right]]>= left) left = lastPos[s[right]] + 1;
            lastPos[s[right]] = right;
            maxLen = max(maxLen, right - left + 1);
        }

        return maxLen;
    }
};

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        maxi = -float('inf')
        def maxSum(node, summ):
            nonlocal maxi
            if not node:
                return 0
            lSum = maxSum(node.left, maxi)
            rSum = maxSum(node.right, maxi)
            maxi = max(maxi, lSum + rSum + node.val)
            return node.val + max(lSum, rSum)
        maxSum(root, 0)
        return maxi
